### PR TITLE
opencv3: enable building of image libs and fix audit errors

### PR DESCRIPTION
**Pull request by rggjan**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### CODE REVIEW

**Comment by iMichka** [comment]
> failed: brew linkage --test visp
Stacktrace
        Missing libraries:
  /usr/local/opt/gsl/lib/libgsl.19.dylib

gal was bumped a few days ago, and visp was updated: #5893

Let me trigger a new mac build.

**Comment by iMichka** [comment]
@BrewTestBot test this please

File: opencv3.rb [hunk 1]
```diff
@@ -136,6 +134,18 @@ def install
     args << "-DWITH_QT=" + (with_qt ? "ON" : "OFF")
     args << "-DWITH_TBB=" + arg_switch("tbb")
     args << "-DWITH_VTK=" + arg_switch("vtk")
+    args << "-DBUILD_TIFF=" + neg_arg_switch("libtiff")
+    args << "-DBUILD_PNG=" + neg_arg_switch("libpng")
+
+    if build.with?("jpg") && build.with?("jpeg-turbo")
```
Comment (iMichka) [inline opencv3.rb]: This should be `jpeg` instead of `jpg`

File: opencv3.rb [hunk 2]
```diff
@@ -136,6 +134,18 @@ def install
...
+      odie "Options --with-jpg and --with-jpeg-turbo are mutually exclusive."
+    elsif build.without?("jpg") && build.without?("jpeg-turbo")
```
Comment (iMichka) [inline opencv3.rb]: Same here, `jpeg` instead of `jpg`

File: opencv3.rb (hunk 1 above)
Comment (rggjan) [inline opencv3.rb]: Of course, my mistake. Fixing it right now...

**Comment by rggjan** [comment]
Fixed jpg->jpeg, and added a few other fixes

File: opencv3.rb (hunk 2 above)
Comment (rggjan) [inline opencv3.rb]: Fixed as well

**Comment by iMichka** [comment]
@rggjan Thanks for your contribution, and thanks for the audit fixes.

**Comment by rggjan** [comment]
@iMichka You're welcome! Thanks for the review and merge :)

**Pull request closed without merging**